### PR TITLE
[libc][NFC] Fix `-DSHOW_INTERMEDIATE_OBJECTS=DEPS` to work properly for entry points and unit tests.

### PR DESCRIPTION
**Pull request by lntue**

### CODE REVIEW

File: libc/cmake/modules/LLVMLibCObjectRules.cmake [hunk 1]
```diff
@@ -652,7 +652,7 @@ function(create_entrypoint_object fq_target_name)
   if(SHOW_INTERMEDIATE_OBJECTS)
     message(STATUS "Adding entrypoint object ${fq_target_name}")
     if(${SHOW_INTERMEDIATE_OBJECTS} STREQUAL "DEPS")
-      foreach(dep IN LISTS ADD_OBJECT_DEPENDS)
+      foreach(dep IN LISTS ADD_ENTRYPOINT_OBJ_DEPENDS)
         message(STATUS "  ${fq_target_name} depends on ${dep}")
```
Comment (SchrodingerZhu) [inline libc/cmake/modules/LLVMLibCObjectRules.cmake:656]: Would this be too verbose? How about just print out a `;` separated list?

Comment (lntue) [reply libc/cmake/modules/LLVMLibCObjectRules.cmake:656]: Yes, in this mode it has been very verbose.  But I tried with just print out the list before, and it was very hard to parse and inspect visually.

**Review by nickdesaulniers — APPROVED** [review]
